GeometrickFX/Shapes/ConcreteDomain/Circle.cs

```diff
@@ -0,0 +1,31 @@
+﻿using Shapes.AbstractDomain;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Shapes.ConcreteDomain
+{
+    public class Circle : IShape
+    {
+        private double _radius;
+        readonly Expression<Func<double, double>> _formula;
+        public Circle(double radius, Expression<Func<double, double>> formula)
```

Comment:
This Expression<Func<double, double>> formula is nice, but I expect the Circle to know how to calculate its area. Also you could be injected and Func there, including malicious ones :)